examples/simple_example.rs

```diff
@@ -0,0 +1,136 @@
+        let alba = CentralizedTelescope::create(params);
+        let proof = alba.prove(&prover_set).unwrap();
+        let signatures = proof.element_sequence.clone();
+        let mut public_keys = Vec::with_capacity(signatures.len());
+
+        for sig in signatures {
+            public_keys.push(
+                *key_list
+                    .get(alba_signatures.get(sig.as_slice()).unwrap())
+                    .unwrap(),
+            );
+        }
+        Self {
+            proof,
+            key_list: public_keys,
+        }
+    }
+
+    /// Validates individual signatures in the threshold signature
+    fn validate_signatures(&self, msg: &[u8]) -> bool {
+        let mut signatures = Vec::with_capacity(self.proof.element_sequence.len());
+        for sig_bytes in &self.proof.element_sequence {
+            if let Ok(signature) = Signature::from_bytes(sig_bytes.as_slice()) {
+                signatures.push(signature);
+            } else {
+                println!("Error: Failed to parse signature from bytes.");
+                return false;
+            }
+        }
+        let signature_refs: Vec<&Signature> = signatures.iter().collect();
+        let aggregate_signature =
+            if let Ok(agg_sig) = AggregateSignature::aggregate(signature_refs.as_slice(), false) {
+                agg_sig.to_signature()
+            } else {
+                println!("Error: Failed to aggregate signatures.");
+                return false;
+            };
+
+        let public_key_refs: Vec<&PublicKey> = self.key_list.iter().collect();
+        let aggregate_public_key =
+            if let Ok(agg_pk) = AggregatePublicKey::aggregate(public_key_refs.as_slice(), false) {
+                agg_pk.to_public_key()
+            } else {
+                println!("Error: Failed to aggregate public keys.");
+                return false;
+            };
+
+        let result = aggregate_signature.verify(false, msg, &[], &[], &aggregate_public_key, false);
+        result == BLST_ERROR::BLST_SUCCESS
+    }
+
+    fn verify(&self, msg: &[u8], params: &Params) -> bool {
+        if self.validate_signatures(msg) {
+            let alba = CentralizedTelescope::create(params);
+            return alba.verify(&self.proof);
+        }
+        false
+    }
+}
+
+fn main() {
+    let mut rng = ChaCha20Rng::from_seed(Default::default());
+    let mut msg = [0u8; 16];
+    rng.fill_bytes(&mut msg);
+    let set_size = 1000;
+    let params = Params {
+        soundness_param: 10.0,
+        completeness_param: 10.0,
+        set_size: 80 * set_size / 100,
+        lower_bound: 20 * set_size / 100,
+    };
+
+    let mut key_list = HashMap::with_capacity(set_size as usize);
+    let mut signature_list = HashMap::with_capacity(set_size as usize);
+
+    for i in 0..set_size as usize {
+        let signer = AlbaSigner::new(&mut rng);
+        key_list.insert(i, signer.verification_key);
+        signature_list.insert(signer.sign::<DATA_LENGTH>(&msg), i);
```

Comment:
Why not having a single hashmap that uses a signature as key and public key as value? I think that would work.